### PR TITLE
Rtd components

### PR DESCRIPTION
**Pull request by ciaradunks**
This PR is created to include the readthedocs documentation for SMOOTH. It includes a 'Getting started' page with four main sections: components, examples, framework and optimization. 

ToDo: include diagrams for run_smooth and run_optimization

### CODE REVIEW

**Review by j-brendel — COMMENTED** [review]
This review only includes getting_started.rst
The Rest is still to come

File: doc/source/getting_started.rst [hunk 1]
```diff
@@ -32,18 +40,70 @@ You also need to install the solver for oemof. This can be done according to
 documentation page.
```
[the remark this answers is not in the document]
Comment (ciaradunks) [reply doc/source/getting_started.rst:40]: Sure, made these changes now

**Comment by j-brendel** [comment]
As for the general structure I think it would make sense to have the idea of what smooth does and what the optimization does somewhere prominent at the beginning. I would propose to add a section in getting started after "Installing smooth" and before "Structure of the smooth modulle" where it is briefly explained. 
It would just be like: 
SMOOTH solves an explicitly defined energy system with several components, such as electrolyzers or a hydrogen storages. The results are the flows in and out of each component as well as their states for all timesteps defined for the simulation and the associated emissions and costs, which are capital expenditures (CAPEX), operational expenditures (OPEX) and variable costs.
The SMOOTH optimization can be used additionally. It finds an optimal energy system by varying key parameters of the components, such as power or capacity and simulating these modified systems using SMOOTH. The optimization than evaluates all individual energy systems according to two objective functions, that can be defined to measure the performance of each system. Finally the specifications of the best suited systems are given as results.

File: smooth/examples/run_optimization_example.py [hunk 2]
```diff
@@ -1,3 +1,72 @@
...
+* whether or not the detailed final results should be saved
+
+For instance, in the below example of the variables being defined, the
+optimization is based on minimizing costs and emissions, evaluating 8
+individuals for 2 generations with the use of the maximum possible
+number of cores available.
+
+.. code:: bash
+
+    opt_params['ga_params'] = {
```
Comment (j-brendel) [inline smooth/examples/run_optimization_example.py:27]: I would like to see 'ignore_zero=True' included

**Review by j-brendel — COMMENTED** [review]
I sent you an email @ciaradunks

File: doc/source/getting_started.rst [hunk 3]
```diff
@@ -14,36 +22,116 @@ SMOOTH stands for "Simulation Model for Optimized Operation and Topology of Hybr
...
+costs/revenues and emissions are transferred to annuities (kg/a and EUR/a, respectively) based on the component lifetimes, and the 
+total system financial and emissions annuities are recorded. The notable states of the components and the energy and mass flows of 
+the system are also recorded and all results can be saved for later use.
+
+An additional functionality of SMOOTH is the optimization (MOEA) which optimizes the topology and operational management of an
+energy system with regards to ecological and economic target criteria. Key parameters of components are chosen, such as the
+maximum power output or capacity, and varied in numerous versions of the energy system until the optimal solution/s is/are 
+reached. The specification of the final system/s is/are finally returned as SMOOTH results.
+
+Structure of the SMOOTH module
```
Comment (j-brendel) [inline doc/source/getting_started.rst:72]: Well written

**Review by j-brendel — APPROVED** [review]
Ready to go. Thanks for that @ciaradunks !

**Pull request merged**